lib/graphql/object/resolvers.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+# test_via: ../object.rb
+
+module GraphQL
+  class Object < GraphQL::SchemaMember
+    module Resolvers
+      class Dynamic
+        def initialize(method_name:)
+          @method_name = method_name
+        end
+
+        def call(obj, args, ctx)
+          if obj.respond_to?(@method_name)
+            public_send_field(obj, @method_name, args)
+          elsif obj.object.respond_to?(@method_name)
```

Comment:
At some point, should we ensure that the only public instance methods on the `object` class correspond to fields?

Reply:
I love the idea, but I also want to reduce friction for people getting started. Maybe the way to do it is a `Schema#lint`-type method that checks for that.